src/Orchard.Web/Modules/Orchard.SecureSocketsLayer/Services/SecureSocketsLayerService.cs

```diff
@@ -226,24 +227,28 @@ public class SecureSocketsLayerService : ISecureSocketsLayerService {
             var settings = GetSettings();
             if (settings == null) return path;
             var secureHostName = settings.SecureHostName;
-            var builder = new UriBuilder(secureHostName.Trim('/') + path) {
+            int port = 443;
+            TryGetPort(secureHostName, port, out port);
+            var builder = new UriBuilder(secureHostName.Split(':').First().Trim('/') + path) {
                 Scheme = Uri.UriSchemeHttps, 
-                Port = 443
+                Port = port
             };
-            SetPort(secureHostName, builder);
             return builder.Uri.ToString();
         }
 
-        private static void SetPort(string hostName, UriBuilder builder) {
-            if (string.IsNullOrWhiteSpace(hostName)) return;
+        private static bool TryGetPort(string hostName, int defaultPort, out int port) {
```

Comment:
It's not a TryXXX pattern, it's just a `int GetPort(string hostName)` which returns either 80 or a port if specified.